### PR TITLE
Configure environment variables for Traefik

### PR DESCRIPTION
**Pull request by lawliet89**
Signed-off-by: Yong Wen Chua <lawliet89@users.noreply.github.com>

#### What this PR does / why we need it:

Add environment variables configuration for Traefik containers.

Use case might include adding an `status.hostIP` environment variable to use the local Consul agent (which is run as `DaemonSet`) when installed with the official HashiCorp helm chart, for example.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/traefik/README.md [hunk 1]
```diff
@@ -105,6 +105,7 @@ The following table lists the configurable parameters of the Traefik chart and t
 | `memoryLimit`                          | Memory limit per Traefik pod                                                                                                 | `30Mi`                                            |
 | `rbac.enabled`                         | Whether to enable RBAC with a specific cluster role and binding for Traefik                                                  | `false`                                           |
 | `deploymentStrategy`                   | Specify deployment spec rollout strategy                                                                                     | `{}`                                              |
+| `deploymentStrategy`                   | Environment variables for the container                                                                                      | `{}`                                              |
```
Comment (cpanato) [inline stable/traefik/README.md]: this is duplicate, maybe this is `env`

**Review by cpanato — CHANGES_REQUESTED** [review]
see comment above.

File: stable/traefik/README.md (hunk 1 above)
Comment (lawliet89) [reply stable/traefik/README.md]: Oops, I've fixed the typo.

**Comment by lawliet89** [comment]
@dtomcej Please review this.

My implementation is different from #10812 in that it allows more flexibility in how you define your environment variables, including the use of the downward API.

I am using Consul and I would like my Traefik pods to use the Consul agent running on each node (as `DaemonSet`) instead of talking directly to the Consul server.

**Comment by rockwe11** [comment]
Hey @lawliet89, could you please fix the conflicts in this PR so it could be merged?
I also interested in your implementation since it is more flexible and allows to use downward API (need to send metrics and traces to Datadog agents deployed as DaemonSet).

**Comment by lawliet89** [comment]
@rockwe11 I've fixed the conflicts.

**Comment by cpanato** [comment]
@lawliet89 there are still conflicts, are you able to update the branch?

**Comment by lawliet89** [comment]
@cpanato That's done.

**Comment by rockwe11** [comment]
Hey @cpanato,
Looks like @lawliet89 made the changes you requested. Could you please review and merge if everything looks good to you?

**Comment by dtomcej** [comment]
@lawliet89 This looks great. Can you rebase, and bump the minor version? (this isn't a patch 😛)

Thanks!

**Comment by lawliet89** [comment]
Done

**Comment by dtomcej** [comment]
Awesome!

**Pull request merged**